src/quo/components/slideshow/slider_bar/view.cljs

```diff
@@ -0,0 +1,162 @@
+        (< relative-index 0) MICRO_SCALE
+        (= relative-index 0) (first-item {:active-index active-index})
+        (= relative-index 1) (second-item {:active-index active-index})
+        (= relative-index 4) (butlast-item {:total-length total-length
+                                            :active-index active-index})
+        (= relative-index 5) (last-item {:total-length total-length
+                                         :active-index active-index})
+        (> relative-index 5) MICRO_SCALE
+        :else
+        DEFAULT_SCALE))))
+
+(defn make-item-label
+  [index active?]
+  (keyword (str "slider-item-" index (when active? "-active"))))
+
+(defn- item
+  [{:keys [index active-index total-length active? customization-color theme blur?]
+    :or   {customization-color :blue
+           theme               :light}}]
+  (let [shared-scale (reanimated/use-shared-value (get-scale index active-index total-length))]
+    (rn/use-mount (fn []
+                    (let [new-scale-value (get-scale index active-index total-length)]
+                      (reanimated/animate shared-scale new-scale-value))))
+    [rn/view
+     {:key   index
+      :style (style/item-wrapper {:size    ITEM_SIZE
+                                  :spacing ITEM_SPACING})}
+     [reanimated/view
+      {:accessibility-label (make-item-label index active?)
+       :style
+       (reanimated/apply-animations-to-style
+        {:transform [{:scale shared-scale}]}
+        (style/item {:size                ITEM_SIZE
+                     :spacing             ITEM_SPACING
+                     :active?             active?
+                     :customization-color customization-color
+                     :theme               theme
+                     :blur?               blur?}))}]]))
+
+(defn- get-item-layout
+  [_ index]
+  (let [length (+ ITEM_SIZE ITEM_SPACING)]
+    #js {:length length
+         :index  index
+         :offset (* index length)}))
+
+(defn- view-internal
+  [{:keys [total-amount active-index customization-color blur? accessibility-label]
+    :or   {customization-color :blue
+           blur?               false
+           active-index        0
+           total-amount        1}}]
+  (let [theme            (quo.theme/use-theme)
+        flat-list-ref    (rn/use-ref-atom nil)
+        on-flat-list-ref (rn/use-callback #(reset! flat-list-ref %) [flat-list-ref])
+        center-position  0.5
+        data             (range total-amount)
+        scroll-to-index  (rn/use-callback
+                          (fn []
+                            (.scrollToIndex ^js @flat-list-ref
+                                            #js {:animated     true
+                                                 :index        active-index
+                                                 :viewOffset   ITEM_SPACING
+                                                 :viewPosition center-position}))
+                          [flat-list-ref active-index])
```

Comment:
Again, I'm wondering if this code is ok or may lead to bugs :thinking: 

What I'd say is:
`on-flat-list-ref` doesn't need a dependency vector. AFAIK, `flat-list-ref` is just an atom, which is a JS object, with a stable reference since our hook `use-ref-atom` doesn't redefine it on each re-render, so it means `on-flat-list-ref` will always work without the dependency vector. Consider this comment for all the dependency vectors containing `flat-list-ref`.

CC: @ilmotta wdyt?
IMO the reasoning behind our custom hooks is becoming unclear sometimes, the dev needs to deeply understand our implementations, leading maybe to bugs :thinking: 

We should consider making a simpler implementation, particularly  `use-ref-atom` returns an atom to `deref` and to `reset`/`swap`, @J-Son89  had an idea of returning something like `[ref set-ref]` which would handle this in a simpler way.

Reply:
Consider renaming `on-flat-list-ref` maybe to `set-flat-list-ref`

Reply:
> on-flat-list-ref doesn't need a dependency vector. AFAIK, flat-list-ref is just an atom, which is a JS object, with a stable reference since our hook use-ref-atom doesn't redefine it on each re-render, so it means on-flat-list-ref will always work without the dependency vector. Consider this comment for all the dependency vectors containing flat-list-ref

That makes sense to me as well @ulisesmac 👍🏼 

> IMO the reasoning behind our custom hooks is becoming unclear sometimes, the dev needs to deeply understand our implementations, leading maybe to bugs

@seanstrom experimented with some interesting ideas to reduce the mental overhead of keeping the dependencies correct. Looking at other Clojure libs, this problem is being attacked by some, also via macros. I believe it's worth giving this problem more seriousness because it will just grow in `status-mobile` and the cost is not negligible.

I don't know about other CCs, but I already wasted time debugging my own code a few times due to missing dependencies, not to mention that's a chore to maintain the dependencies as the code evolves.

On top of that, now I have to manually re-test carefully all cached functions to see if they work well after re-renders.